repo/currency_definition.go

```diff
@@ -336,8 +374,8 @@ func (c *CurrencyDefinition) Valid() error {
 
 // Equal indicates if the receiver and other have the same code
 // and divisibility
-func (c *CurrencyDefinition) Equal(other *CurrencyDefinition) bool {
-	if c == nil || other == nil {
+func (c CurrencyDefinition) Equal(other CurrencyDefinition) bool {
+	if c == NilCurrencyDefinition || other == NilCurrencyDefinition {
```

Comment:
Not that this is an expected usage, but should the case where both c and other are NilCurrencyDefinition return true?